### DISCUSSION TITLE
Coroutine 'lifespan' not awaited

### DISCUSSION THREAD

**Discussion by daudln** [fastapi/fastapi#11375] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from contextlib import asynccontextmanager

from faker import Faker
from fastapi import FastAPI

@asynccontextmanager
async def app_lifespan(app: FastAPI):
    for _ in range(100):
        await User.create(
            name=faker.name(),
            email=faker.email(),
        )

app = FastAPI(lifespan=app_lifespan)

register_tortoise(
    app,
    db_url="sqlite://data.sqlite3",
    modules={"models": [__name__]},
    generate_schemas=True,
)


if __name__ == "__main__":
    uvicorn.run("testapp:app", port=7000, reload=True)
```

### Description

I'm encountering an error with FastAPI's lifespan coroutine

I'm trying to use the lifespan functionality in a FastAPI application to perform some startup tasks, like creating users in this case. However, I'm running into the error `"coroutine 'lifespan' was never awaited"` when running the application with uvicorn.

## Expected Behavior

I expect uvicorn to automatically await the app_lifespan coroutine, which would trigger the defined startup logic before the application starts handling requests. This would ensure tasks like user creation happen before the app is live.

## Current Behavior

The error message indicates that uvicorn is not awaiting the app_lifespan coroutine.

### Operating System

Linux

### Operating System Details

Ubuntu 23.10

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.10.9

**Comment by Kludex** [comment, marked as answer]
Missing yield on lifespan.

**Reply by daudln** [reply]
When I add yield statement like this

```python
@asynccontextmanager
async def app_lifespan(app: FastAPI):
    for _ in range(100):
        await User.create(
            name=faker.name(),
            email=faker.email(),
        )
    yield
```

TypeError: 'NoneType' object is not iterable #6400 which I answered it, and goes perfect if app_lifespan is not applied to FastAPI app initialization

**Reply by Kludex** [reply]
Can you reproduce this without tortoise?

**Reply by daudln** [reply]
No!
Thank you, I think this issue is for tortoise integration with FastAPI, the app is running after commenting the tortoise ORM logic.

```py
@asynccontextmanager
async def app_lifespan(app: FastAPI):
    # for _ in range(100):
    #     await User.create(
    #         name=faker.name(),
    #         email=faker.email(),
    #     )
    yield
```

**Reply by daudln** [reply]
I have reported it to Tortoise ORM repo [tortoise-orm/issues/1576](https://github.com/tortoise/tortoise-orm/issues/1576) may be they can fix it.
